lib/h3core.js

```diff
@@ -597,6 +597,12 @@ export function h3ToParent(h3Index, res) {
  * @return {H3Index[]}        H3 indexes of children, or empty array for invalid input
  */
 export function h3ToChildren(h3Index, res) {
+    // TODO: There appears to be a potential infinite loop on bad input here.
+    // Validate and return an empty array for now; update this behavior when
+    // the C lib handles the bad input more appropriately.
```

Comment:
That seems very likely - in that case the C code may be behaving as expected. It may simply be a particularly bad choice of invalid input. This still makes me nervous, however, and I think the validity check may still be warranted. Thoughts?

Reply:
I'd prefer the validity check being in. You may want to update the comment though.

Reply:
FYI - `parseInt("foo", 16)` results in `15` (parses the `"f"` and then stops parsing). In binary that's 28 leading zeros and 4 ones, so it might look like a res 0 cell to a liberal algorithm.

Reply:
Agreed. Updated the comment to better explain the situation.